src/logic/use-stop-place-with-estimated-calls/types.ts

```diff
@@ -0,0 +1,79 @@
+import {
+    array,
+    boolean,
+    number,
+    string,
+    type,
+    Infer,
+    is,
+    nullable,
+} from 'superstruct'
+import {
+    DateStruct,
+    DateTimeStruct,
+    TransportModeEnumStruct,
+    TransportSubmodeEnumStruct,
+} from '../../utils/structs'
+
+const SituationStruct = type({
+    summary: array(
+        type({
+            value: string(),
+        }),
+    ),
+})
+
+const EstimatedCallStruct = type({
+    aimedDepartureTime: DateTimeStruct,
+    cancellation: boolean(),
+    date: DateStruct,
+    destinationDisplay: type({
+        frontText: string(),
+    }),
+    expectedDepartureTime: DateTimeStruct,
+    quay: type({
+        id: string(),
+        name: string(),
+        publicCode: nullable(string()),
+    }),
+    serviceJourney: type({
+        id: string(),
+        journeyPattern: type({
+            line: type({
+                publicCode: nullable(string()),
+                transportMode: TransportModeEnumStruct,
+            }),
+        }),
+        transportSubmode: TransportSubmodeEnumStruct,
+    }),
+    situations: array(SituationStruct),
+})
+
+const StopPlaceWithEstimatedCallsStruct = type({
+    id: string(),
+    name: string(),
+    description: nullable(string()),
+    latitude: number(),
+    longitude: number(),
+    transportMode: array(TransportModeEnumStruct),
+    transportSubmode: array(TransportSubmodeEnumStruct),
+    estimatedCalls: array(EstimatedCallStruct),
```

Comment:
Ikke helt stødig på superstruct enda, vil sjekken for hele objektet feile om ett medlem av en array feiler? Hvis det er tilfellet kan f.eks. en "null"-situation få hele objektet til å bli invalid som virker litt spooky.

Reply:
Ja, hele sjekken vil feile om et element feiler eller er null. Det er ganske strengt ja, men er usikker på om alternativet er mye bedre. Vi kan skrive `array(nullable(EstimatedCallStruct))`, men da må vi gjøre en filter på listen etterpå. Det er ikke så big deal, men jeg tenker at det er en så rar situasjon at et element er null at det kanskje er best å bare feile hardt 🤔